veps/sig-compute/60-individual-and-multifunction-hostdevices/individual-and-multifunction-hostdevices.md

```diff
@@ -0,0 +1,352 @@
+
+During single-function passthrough, a sequence of events occurs within the Linux kernel that completely cuts power to the PCIe slot. *In short, QEMU triggers Linux (VFIO) to reset each function and if possible it will issue a full slot or bus reset*  
+
+When dealing with multifunction devices, a Kubernetes node may have multiple instances of the same multifunction PCI device. If a VirtualMachine requests two function devices, the scheduler may allocate one function from two different physical devices residing for example on two separate pcie slots.  
+
+If this occurs, Linux will not cut power to the host device during passthrough, resulting in behavioral discrepancies compared to scenarios where both functions originate from the same device. *This difference in power state transitions can impact device initialization for certain hardware.*  
+
+Passing the entire host device including all of its functions allows KubeVirt to be used in automated testing farms of pci host devices.  
+This allows CI jobs and PCI driver developers to request a virtual machine with a host device and gain access to the entire device.  
+For PCI driver development, it's crucial to boot a VM with a preserved PCI device tree mirroring the host system and to perform a bus reset prior to VM startup.  
+
+## Goals  
+
+- Allow booting a VM using an entire PCI device with all its associated functions.  
+
+## Non-Goals  
+
+- Support for live migration.  
+- Preserve the PCI tree topology in the guest.  
+
+## Definition of Users  
+
+- VM owners.  
+- cluster admins.  
+
+## User Stories  
+
+- As a VM owner, I want the ability to allocate an entire PCI device, including all its functions, to my VM.  
+- As a cluster admin, I want a robust solution for generic host device PCI passthrough to VMs.  
+
+## Repositories Impacted  
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)    
+
+## Design  
+
+**Prerequisites:** To understand this enhancement, familiarity with Kubernetes' Dynamic Resource Allocation (DRA) framework and its implementation in KubeVirt is required.
+- [DRA in Kubernetes](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/)
+- [Existing DRA implementation in KubeVirt](../10-dra-devices/vep.md)
+
+The existing implementation of DRA in KubeVirt expects a 3rd party DRA driver to be installed in the cluster.  
+This driver manages device binding to `vfio` and publishes discovered devices to `ResourceSlice` objects.  
+Once a `ResourceSlice` is populated, a `ResourceClaim` object can be created and referenced in the `VirtualMachine` spec.
+
+Currently, KubeVirt does not provide an official DRA driver for generic PCI passthrough.
+
+This enhancement proposal focuses on the following:
+- Proposing a design for a generic DRA driver for PCI devices maintained by KubeVirt.
```

Comment:
How do we differentiate what kind of hardware is supported by the generic DRA driver and what needs vendor specific DRA driver?